functions/Set-DbaTcpPort.ps1

```diff
@@ -116,7 +122,17 @@ function Set-DbaTcpPort {
                 if ($Pscmdlet.ShouldProcess($instance, "Setting port to $Port for IP address $IpAddress of $instance")) {
                     $netConf | Set-DbaNetworkConfiguration -Credential $Credential -EnableException:$EnableException -Confirm:$false
                 }
+
+                if (Test-Bound -ParameterName Force) {
+                    if ($PSCmdlet.ShouldProcess($instance, "Force provided, restarting Engine and Agent service for $instance on $computerFullName")) {
+                        try {
+                            $null = Restart-DbaService -ComputerName $computerFullName -InstanceName $instanceName -Type Agent, Engine
```

Comment:
Please only restart the Engine. The Agent would be restarted if needed.

You could also use the new parameter -SqlInstance with $instance.

But in order to use the try-catch, you have to use -EnableException.

And thanks for the hint about DbaAgHadr - I will take a look and fix it there and in some more places...

Reply:
Is -Force required to restart the agent without prompting?

Reply:
Yes, you are right. So this has to be added as well.

Reply:
So it might be one option: `$null = Restart-DbaService -SqlInstance $instance -Type Engine -Force -EnableException`.

Reply:
Thanks for the tips ya'll! Hard to keep up with how often you improve these patterns :D